makelib/controlplane.mk

```diff
@@ -1,41 +0,0 @@
-# Copyright 2022 The Upbound Authors. All rights reserved.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-KIND_CLUSTER_NAME ?= local-dev
-CROSSPLANE_NAMESPACE ?= upbound-system
-
-CONTROLPLANE_DUMP_DIRECTORY ?= $(OUTPUT_DIR)/controlplane-dump
-
-controlplane.up: $(UP) $(KUBECTL) $(KIND)
-	@$(INFO) setting up controlplane
-	@$(KIND) get kubeconfig --name $(KIND_CLUSTER_NAME) >/dev/null 2>&1 || $(KIND) create cluster --name=$(KIND_CLUSTER_NAME)
-	@$(INFO) "setting kubectl context to kind-$(KIND_CLUSTER_NAME)"
-	@$(KUBECTL) config use-context "kind-$(KIND_CLUSTER_NAME)"
-ifndef CROSSPLANE_ARGS
-	@$(INFO) setting up crossplane core without args
-	@$(KUBECTL) -n $(CROSSPLANE_NAMESPACE) get cm universal-crossplane-config >/dev/null 2>&1 || $(UP) uxp install $(UXP_VERSION) --namespace=$(CROSSPLANE_NAMESPACE) $(UXP_INSTALL_OPTS)
```

Comment:
@ulucinar @turkenh I realized after deleting this that Upjet providers are using this file and `local.xpkg.mk` to run uptest. I'm guessing that means we need to keep those two files if I want upjet providers to switch to this submodule. 🙂 

I can revert the deletes so that Upjet providers can switch over to the `crossplane/build` submodule, but I notice this file is using UXP specifically. Now that we donated upjet to Crossplane, could I update this to use upstream Crossplane instead?

Reply:
Here's the link to where it's used: https://github.com/crossplane/upjet-provider-template/blob/65f168d69b3a403d022f822e16439447c9ebe48e/Makefile#L194

Reply:
I'm going to guess these are needed and reintroduce them. 😉

Reply:
Yes. There were used to start a controlplane (kind + crossplane) for all uptest base CIs/targets. 
Thanks for catching that.